### ISSUE TITLE
Bus error (core dumped)

### ISSUE THREAD

**Issue by flysssss** [rmurai0610/MASt3R-SLAM#67]
conda install pytorch==2.5.1 torchvision==0.20.1 torchaudio==2.5.1  pytorch-cuda=11.8 -c pytorch -c nvidia

error:

(mast3r-slam) root@ys-ai-GPU-7:/tmp/45test3d/code/MASt3R-SLAM# python main.py --dataset datasets/tum/rgbd_dataset_freiburg1_room/ --config config/calib.yaml
Warning, cannot find cuda-compiled version of RoPE2D, using a slow pytorch version instead
datasets/tum/rgbd_dataset_freiburg1_room/
{'use_calib': True, 'single_thread': False, 'dataset': {'subsample': 2, 'img_downsample': 1, 'center_principle_point': True}, 'matching': {'max_iter': 10, 'lambda_init': 1e-08, 'convergence_thresh': 1e-06, 'dist_thresh': 0.1, 'radius': 3, 'dilation_max': 5}, 'tracking': {'min_match_frac': 0.05, 'max_iters': 50, 'C_conf': 0.0, 'Q_conf': 1.5, 'rel_error': 0.001, 'delta_norm': 0.001, 'huber': 1.345, 'match_frac_thresh': 0.333, 'sigma_ray': 0.003, 'sigma_dist': 10.0, 'sigma_pixel': 1.0, 'sigma_depth': 10.0, 'sigma_point': 0.05, 'pixel_border': -10, 'depth_eps': 1e-06, 'filtering_mode': 'weighted_pointmap', 'filtering_score': 'median'}, 'local_opt': {'pin': 1, 'window_size': 1000000.0, 'C_conf': 0.0, 'Q_conf': 1.5, 'min_match_frac': 0.1, 'pixel_border': -10, 'depth_eps': 1e-06, 'max_iters': 10, 'sigma_ray': 0.003, 'sigma_dist': 10.0, 'sigma_pixel': 1.0, 'sigma_depth': 10.0, 'sigma_point': 0.05, 'delta_norm': 1e-08, 'use_cuda': True}, 'retrieval': {'k': 3, 'min_thresh': 0.005}, 'reloc': {'min_match_frac': 0.3, 'strict': True}, 'inherit': 'config/base.yaml'}
Warning, cannot find cuda-compiled version of RoPE2D, using a slow pytorch version instead
Bus error (core dumped)

**Comment by flysssss** [comment]
(mast3r-slam) root@ys-ai-GPU-7:/tmp/45test3d/code/MASt3R-SLAM# pip list
Package                   Version      Editable project location
------------------------- ------------ ------------------------------------------------
absl-py                   2.2.0
aiofiles                  23.2.1
annotated-types           0.7.0
anyio                     4.9.0
argcomplete               3.6.1
asmk                      0.1
attrs                     25.3.0
Brotli                    1.0.9
certifi                   2025.1.31
charset-normalizer        3.3.2
click                     8.1.8
colorama                  0.4.6
colorlog                  6.9.0
contourpy                 1.3.1
curope                    0.0.0
cycler                    0.12.1
einops                    0.8.1
embreex                   2.17.7.post6
evo                       1.31.1
faiss-cpu                 1.10.0
fastapi                   0.115.12
ffmpy                     0.5.0
filelock                  3.13.1
fonttools                 4.56.0
fsspec                    2025.3.0
glcontext                 3.0.0
glfw                      2.8.0
gmpy2                     2.2.1
gradio                    5.23.1
gradio_client             1.8.0
groovy                    0.1.2
grpcio                    1.71.0
h11                       0.14.0
httpcore                  1.0.7
httpx                     0.28.1
huggingface-hub           0.29.3
idna                      3.7
imgui                     2.0.0
in3d                      0.0.1        /tmp/45test3d/code/MASt3R-SLAM/thirdparty/in3d
Jinja2                    3.1.6
joblib                    1.4.2
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
kiwisolver                1.4.8
lietorch                  0.2
lxml                      5.3.1
lz4                       4.4.3
manifold3d                3.0.1
mapbox_earcut             1.0.3
Markdown                  3.7
markdown-it-py            3.0.0
MarkupSafe                3.0.2
MAST3R                    0.0.1        /tmp/45test3d/code/MASt3R-SLAM/thirdparty/mast3r
MAST3R-SLAM               0.0.1        /tmp/45test3d/code/MASt3R-SLAM
matplotlib                3.10.1
mdurl                     0.1.2
mkl_fft                   1.3.11
mkl_random                1.2.8
mkl-service               2.4.0
moderngl                  5.12.0
moderngl-window           2.4.6
mpmath                    1.3.0
msgpack                   1.1.0
multipledispatch          1.0.0
natsort                   8.4.0
networkx                  3.4.2
numexpr                   2.10.2
numpy                     1.26.4
opencv-python             4.11.0.86
orjson                    3.10.16
packaging                 24.2
pandas                    2.2.3
pillow                    11.1.0
pip                       25.0
plyfile                   1.1
protobuf                  6.30.1
pyaml                     25.1.0
pycollada                 0.9
pydantic                  2.10.6
pydantic_core             2.27.2
pydub                     0.25.1
pyglet                    2.1.3
pyglm                     2.8.0
Pygments                  2.19.1
pyparsing                 3.2.3
pyrealsense2              2.55.1.6486
pyrr                      0.10.3
PySocks                   1.7.1
python-dateutil           2.9.0.post0
python-multipart          0.0.20
pytz                      2025.2
PyYAML                    6.0.2
referencing               0.36.2
requests                  2.32.3
rich                      13.9.4
roma                      1.5.2.1
rosbags                   0.10.9
rpds-py                   0.23.1
rtree                     1.4.0
ruamel.yaml               0.18.10
ruamel.yaml.clib          0.2.12
ruff                      0.11.2
safehttpx                 0.1.6
safetensors               0.5.3
scikit-learn              1.6.1
scipy                     1.15.2
seaborn                   0.13.2
semantic-version          2.10.0
setuptools                75.8.0
shapely                   2.0.7
shellingham               1.5.4
six                       1.17.0
sniffio                   1.3.1
starlette                 0.46.1
svg.path                  6.3
sympy                     1.13.1
tensorboard               2.19.0
tensorboard-data-server   0.7.2
threadpoolctl             3.6.0
tomlkit                   0.13.2
torch                     2.5.1
torchaudio                2.5.1
TorchCodec                0.1.0
torchvision               0.20.1
tqdm                      4.67.1
trimesh                   4.6.5
triton                    3.1.0
typer                     0.15.2
typing_extensions         4.12.2
tzdata                    2025.2
urllib3                   2.3.0
uvicorn                   0.34.0
vhacdx                    0.0.8.post2
websockets                15.0.1
Werkzeug                  3.1.3
wheel                     0.45.1
xatlas                    0.0.9
xxhash                    3.5.0
zstandard                 0.23.0

**Comment by Quinnvanderschaar1** [comment]
I have same issue, did you solve it?

**Comment by jwLee9805** [comment]
I encountered the same error when running in a Docker environment. 
After investigating, I identified it was related to memory limitations within the container.
The solution was to add the --ipc=host flag to my Docker run command.

**Comment by suayu** [comment]
docker do not support UI. You can use x11 server
